src/App.jsx

```diff
@@ -1,66 +1,29 @@
+import '@reshuffle/code-transform/macro';
 import React, { Component } from 'react';
 import { Route, Switch, withRouter } from 'react-router-dom';
-
 import PrivateRoute from './components/PrivateRoute';
-
+import { AuthContext } from '@reshuffle/react-auth';
 import DevSite from './containers/DevSite';
 import Auth from './containers/Auth';
 import Editor from './containers/Editor';
 import Admin from './containers/Admin';
 
 class App extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { userToken: undefined };
-  }
-
-  async componentDidMount() {
-    // if the token doesn't exist, it will be set to null
-    this.setState({ userToken: localStorage.getItem('userToken') });
-  }
-
-  /**
-   * Store a user token for future sessions
-   */
-  storeUserToken = (userToken) => {
-    localStorage.setItem('userToken', userToken);
-    this.setState({ userToken });
-  }
+  static contextType = AuthContext;
 
   render() {
-    const childProps = {
-      userToken: this.state.userToken,
-      storeUserToken: this.storeUserToken,
-    };
-
-    // If the route is an auth or admin route, display it.
+    // If the route is an auth or admin route, display it if user is right REACT_APP_VALID_HOSTED_DOMAIN .
     // Otherwise, defer to the DevSite nested routing
-    if (this.state.userToken !== undefined) {
-      return (
-        <Switch>
-          <PrivateRoute exact path='/editor'
-                        component={Editor}
-                        userToken={childProps.userToken}
-                        {...this.props}
-                        {...childProps}
-          />
-          <PrivateRoute path='/admin'
-                        component={Admin}
-                        userToken={childProps.userToken}
-                        {...this.props}
```

Comment:
I don't think should be removed. Maybe it can be but it should be done in a separate PR.

Reply:
childProps was only holding user tokens so I figured child props wasn't needed but will add back if needed.